client/app/bundles/course/assessment/question/multiple_responses.js

```diff
@@ -0,0 +1,32 @@
+require('jquery-ui/ui/widgets/sortable');
+
+$(document).ready(() => {
+  // When items are removed, there might be gaps in the weights.
+  function update_weights() {
+    $('.weight input').val(index => index + 1);
+  }
+
+  $('.weight').hide();
+  $('.reorder').show();
+
+  $('.multiple-response-options tbody').on('cocoon:after-insert', function(e, insertedItem) {
```

Comment:
I think you can combine them (and remove the params if not required):
```js
$('.multiple-response-options tbody').on('cocoon:after-insert', 'cocoon:after-remove', function() {
  update_weights();
});
```

http://api.jquery.com/on/